rts/Sim/Units/Scripts/CobFile.cpp

```diff
@@ -124,19 +124,25 @@ CCobFile::CCobFile(CFileHandler& in, const std::string& scriptName)
 	scriptLengths.reserve(ch.NumberOfScripts);
 	pieceNames.reserve(ch.NumberOfPieces);
 
+	// code accessor so we don't need to process code area yet, while being able to access its contents
+	int* preCode = reinterpret_cast<int*>(&cobFileData[ch.OffsetToScriptCode]);
+
 	for (int i = 0; i < ch.NumberOfScripts; ++i) {
 		int ofs = *(int *) &cobFileData[ch.OffsetToScriptNameOffsetArray + i * 4];
 		swabDWordInPlace(ofs);
 		scriptNames.emplace_back(reinterpret_cast<const char*>(&cobFileData[ofs]));
 
+		ofs = *(int *) &cobFileData[ch.OffsetToScriptCodeIndexArray + i * 4];
+		swabDWordInPlace(ofs);
+
 		if (scriptNames[scriptNames.size() - 1].find("lua_") == 0) {
 			luaScripts.emplace_back(scriptNames[scriptNames.size() - 1].c_str() + sizeof("lua_") - 1);
+		} else if (swabDWord(*(preCode + ofs)) == 0x10090000 /* SIGNATURE_LUA */) {
```

Comment:
I don't know COB that well, but just curious, what is the advantage of this one compared to the functions prefixed with `lua_` ?